stock_barcodes/wizard/stock_barcodes_read_inventory.py

```diff
@@ -1,93 +1,95 @@
-# Copyright 2019 Sergio Teruel <sergio.teruel@tecnativa.com>
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# Copyright 2023 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 from odoo import _, api, fields, models
-from odoo.exceptions import ValidationError
-from odoo.fields import first
 
 
 class WizStockBarcodesReadInventory(models.TransientModel):
     _name = "wiz.stock.barcodes.read.inventory"
     _inherit = "wiz.stock.barcodes.read"
     _description = "Wizard to read barcode on inventory"
+    _allowed_product_types = ["product"]
 
-    inventory_id = fields.Many2one(comodel_name="stock.inventory", readonly=True)
+    # Overwrite is needed to take into account new domain values
+    product_id = fields.Many2one(domain=[("type", "in", _allowed_product_types)])
     inventory_product_qty = fields.Float(
         string="Inventory quantities", digits="Product Unit of Measure", readonly=True
     )
-    display_inventory_ids = fields.Many2many(
-        comodel_name="stock.inventory", compute="_compute_display_inventory_ids"
+    inventory_quant_ids = fields.Many2many(
+        comodel_name="stock.quant", compute="_compute_inventory_quant_ids"
     )
 
-    def _compute_display_inventory_ids(self):
-        for rec in self:
-            rec.display_inventory_ids = rec.inventory_id.ids
-
-    def name_get(self):
-        return [
-            (
-                rec.id,
-                "{} - {} - {}".format(
-                    _("Barcode reader"), rec.inventory_id.name, self.env.user.name
-                ),
+    def _compute_inventory_quant_ids(self):
+        for wiz in self:
+            quants = self.env["stock.quant"].search(
+                [
+                    ("user_id", "=", self.env.user.id),
+                    "|",
+                    ("inventory_quantity_set", "=", True),
+                    ("inventory_date", "<=", fields.Date.context_today(self)),
+                ],
```

Comment:
These 3 fields user_id, inventory_quantity_set and inventory_date don't exist on stock.quant on odoo v14 !!! These fields exists in v15 but not in v14.